mount/titus-mount-lustre.c

```diff
@@ -0,0 +1,359 @@
+#define _GNU_SOURCE
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+/* getaddrinfo */
+#include <arpa/inet.h>
+#include <netdb.h>
+#include <string.h>
+/* setns */
+#include <sched.h>
+/* for mount syscalls */
+#include <asm-generic/unistd.h>
+#include <linux/mount.h>
+/* fcntl */
+#include "scm_rights.h"
+#include <assert.h>
+#include <fcntl.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+#include "common.h"
+
+#define E(x)                                                                   \
+	do {                                                                   \
+		if ((x) == -1) {                                               \
+			perror(#x);                                            \
+			exit(1);                                               \
+		}                                                              \
+	} while (0)
+
+#define IP_ADDRESS_LEN 256
+
+/*
+ *Construct lustre device mount options in the format expected by the syscall
+ */
+static void add_device_options(const char *ipaddress, const char *mount_name, char *final_options)
+{
+	strcat(final_options, ",device=");
+	strcat(final_options, ipaddress);
+	strcat(final_options, "@tcp:/");
+    strcat(final_options, mount_name);
```

Comment:
m-m-m-meganitpick: indentation (here and at L302)

Reply:
Haha. We have a `make fmt`, let me run that...